crates/revm/src/evm_impl.rs

```diff
@@ -353,6 +347,19 @@ impl<'a, GSPEC: Spec + 'static, DB: Database> Transact<DB::Error> for EVMImpl<'a
         // main return
         handler.main_return(data, call_result, output, &gas)
     }
+
+    #[inline]
+    fn transact(&mut self) -> EVMResult<DB::Error> {
+        #[cfg(not(feature = "optimism"))]
+        {
+            crate::handler::mainnet::default_transact(self)
+        }
+
+        #[cfg(feature = "optimism")]
+        {
+            crate::handler::optimism::default_transact(self)
+        }
+    }
```

Comment:
This should not go here, but inside `transact_preverified` logic will be skipped otherwise.